app/controllers/admin/routes.py

```diff
@@ -157,6 +157,14 @@ def eventDisplay(eventId):
 
     eventData = model_to_dict(event, recurse=False)
     associatedAttachments = AttachmentUpload.select().where(AttachmentUpload.event == event)
+    
+    picurestype = [".jpeg", ".png", ".gif", ".jpg", ".svg", ".webp"]
+    for attachment in associatedAttachments:
+        
+        for extension in picurestype:
+            if (attachment.fileName.endswith(extension)):
+                print(attachment.fileName)
+                imageName = attachment.fileName
```

Comment:
By putting this inside the for loops, you are not accounting for if there is no image so if there is no image  the page breaks, event if there is another kind of attachment.